src/lib.rs

```diff
@@ -114,34 +115,34 @@ pub trait PolynomialCommitment<F: Field>: Sized {
     /// for the polynomial commitment scheme.
     fn setup(
         max_degree: usize,
-    ) -> Result<Self::UniversalParams, Self::Error>;
+    ) -> Result<Self::Parameters, Self::Error>;
 
     /// Constructs public parameters when given as input the maximum degree `degree`
     /// for the polynomial commitment scheme from given seed
     fn setup_from_seed(
         max_degree: usize,
         seed: &[u8],
-    ) -> Result<Self::UniversalParams, Self::Error>;
-
-        /// Specializes the public parameters for polynomials up to the given `supported_degree`
-    /// and for enforcing degree bounds in the range `1..=supported_degree`.
-    fn trim(
-        pp: &Self::UniversalParams,
-        supported_degree: usize,
-    ) -> Result<(Self::CommitterKey, Self::VerifierKey), Self::Error>;
+    ) -> Result<Self::Parameters, Self::Error>;
+
+    /// The low-level single segment single poly commit function.
+    /// Create a dlog commitment to `scalars` using the commitment key `comm_key`.
+    /// Optionally, randomize the commitment using `hiding_generator` and `randomizer`.
+    fn commit(
+        comm_key: &[G],
```

Comment:
As we do not serve a high-level commit function in the trait, we should demand ``comm_key`` to be of type ``CommitterKey``.

Reply:
Seems for now - no, we cannot proceed in this way. "commit" function is used not only with Commiter/Verifiery key. For example in succinct_verify function we have such call:

        let check_commitment_elem: G::Projective = Self::commit(
            &[proof.final_comm_key.clone(), h_prime],
            &[proof.c.clone(), v_prime],
            None,
            None,
        )?;

Reply:
This is the check on the degree 0 instance after all the reduction steps. It seems to that from the formal point of view,  it'd be more consistent to define with G_final a committer/verifier key and change the declaration of commit.

Reply:
The same applies to the committer keys at each step of the dlog reduction.

Reply:
The same applies to the committer keys at each step of the dlog reduction.

Reply:
I've found how to deal with this. Now "commit" function is more generic and takes "commiter key" as a parameter instead of slice. And there is "inner_commit" function added which works on lower level with slices. Please check the actual codebase if such solution is fine